nexus/db-queries/src/db/datastore/console_session.rs

```diff
@@ -154,4 +160,52 @@ impl DataStore {
                 ))
             })
     }
+
+    /// List console sessions for a specific user
+    pub async fn silo_user_session_list(
+        &self,
+        opctx: &OpContext,
+        authn_list: authz::SiloUserAuthnList,
+        pagparams: &DataPageParams<'_, Uuid>,
+    ) -> ListResultVec<ConsoleSession> {
+        opctx.authorize(authz::Action::ListChildren, &authn_list).await?;
+
+        let user_id = authn_list.silo_user().id();
+
+        use nexus_db_schema::schema::console_session::dsl;
+        paginated(dsl::console_session, dsl::id, &pagparams)
+            .filter(dsl::silo_user_id.eq(user_id))
+            // TODO: unlike with tokens, we do not have expiration time here,
+            // so we can't filter out expired sessions by comparing to now. In
+            // the authn code, this works by dynamically comparing the created
+            // and last used times against now + idle/absolute TTL. We may
+            // have to do that here but it's kind of sad. It might be nicer to
+            // make sessions work more like tokens and put idle and absolute
+            // expiration time right there in the table at session create time.
```

Comment:
This is the major outstanding issue. I will probably make the fix a separate PR on top of this one because it's an actual change, not just an addition.